### PR TITLE
xteve: init at 2.2.0.200

### PR DESCRIPTION
**Pull request by NRHelmi**
###### Description of changes
I nix packaged `xTeVe` and added it to my plex setup, I thought that this might be useful to add it to `nixpkgs`.
`xTeVe` is an M3U proxy server for Plex DVR and Emby Live TV

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by JamieMagee** [comment]
Result of `nixpkgs-review pr 207509` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xteve</li>
  </ul>
</details>

File: pkgs/servers/xteve/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+let
+  owner = "xteve-project";
+  repo  = "xTeVe";
+  version = "2.2.0.200";
```
Comment (AndersonTorres) [inline pkgs/servers/xteve/default.nix:6]: Move this inside the `buildGoModule` expression.

File: pkgs/servers/xteve/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    rev = version;
+    sha256 = "sha256-hD4GudSkGZO41nR/CgcMg/SqKjpAO1yJDkfwa8AUges=";
+  };
+
+  vendorSha256 = "sha256-oPkSWpqNozfSFLIFsJ+e2pOL6CcR91YHbqibEVF2aSk=";
+
+  meta = with lib; {
+    description = ''
+      M3U Proxy for Plex DVR and Emby Live TV.
+    '';
```
Comment (AndersonTorres) [inline pkgs/servers/xteve/default.nix:23]:
```suggestion
    description = "M3U Proxy for Plex DVR and Emby Live TV";
```

File: pkgs/servers/xteve/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  version = "2.2.0.200";
+
+in buildGoModule rec {
+  pname = "xteve";
+  inherit version;
+
+  src = fetchFromGitHub {
+    inherit owner repo;
+    rev = version;
+    sha256 = "sha256-hD4GudSkGZO41nR/CgcMg/SqKjpAO1yJDkfwa8AUges=";
```
Comment (AndersonTorres) [inline pkgs/servers/xteve/default.nix:15]:
```suggestion
    hash = "sha256-hD4GudSkGZO41nR/CgcMg/SqKjpAO1yJDkfwa8AUges=";
```

File: pkgs/servers/xteve/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  vendorSha256 = "sha256-oPkSWpqNozfSFLIFsJ+e2pOL6CcR91YHbqibEVF2aSk=";
+
+  meta = with lib; {
+    description = ''
+      M3U Proxy for Plex DVR and Emby Live TV.
+    '';
+    homepage = "https://github.com/xteve-project/xTeVe";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (AndersonTorres) [inline pkgs/servers/xteve/default.nix:26]: Usually we require that you (aka the package creator) list yourself as maintainer here.

However, I will suppose you are not interested on take this role. 
In this case, for the sake of completeness, please add myself (`AndersonTorres`) as maintainer.

File: pkgs/servers/xteve/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+let
+  owner = "xteve-project";
+  repo  = "xTeVe";
+  version = "2.2.0.200";
+
+in buildGoModule rec {
+  pname = "xteve";
+  inherit version;
```
Comment (AndersonTorres) [inline pkgs/servers/xteve/default.nix:10]:
```suggestion
  version = "2.2.0.200";
```

File: pkgs/servers/xteve/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchFromGitHub {
+    inherit owner repo;
```
Comment (AndersonTorres) [inline pkgs/servers/xteve/default.nix:13]:
```suggestion
    owner = "xteve-project";
    repo  = "xTeVe";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -10282,6 +10282,12 @@
     githubId = 34083928;
     name = "Tim DeHerrera";
   };
+  nrhelmi = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:10285]: Isolate this modification in a separate commit. No need for a new PR.

File: pkgs/servers/xteve/default.nix [hunk 8]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildGoModule, fetchFromGitHub }:
```
Comment (AndersonTorres) [inline pkgs/servers/xteve/default.nix:1]: ... and join the rest in the next PR, with the title:

`xteve: init at 2.2.0.200`

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/servers/xteve/default.nix [hunk 9]
```diff
@@ -1,28 +1,22 @@
...
 
   meta = with lib; {
-    description = ''
-      M3U Proxy for Plex DVR and Emby Live TV.
-    '';
+    description = "M3U Proxy for Plex DVR and Emby Live TV";
     homepage = "https://github.com/xteve-project/xTeVe";
     license = licenses.mit;
-    maintainers = with maintainers; [ ];
+    maintainers = with maintainers; [ nrhelmi AndersonTorres ];
```
Comment (AndersonTorres) [inline pkgs/servers/xteve/default.nix:20]: Squash/merge this commit, **with this exact title**:

```xteve: init at 2.2.0.200```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Now squash the commits into two - one for the new maintainer, other for the new package.

**Comment by NRHelmi** [comment]
@AndersonTorres can you give it another check ? BTW is there any documentation on steps (commit policies) to add a new package to nixpkgs ?

File: pkgs/servers/xteve/default.nix [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    hash = "sha256-hD4GudSkGZO41nR/CgcMg/SqKjpAO1yJDkfwa8AUges=";
+  };
+
+  vendorSha256 = "sha256-oPkSWpqNozfSFLIFsJ+e2pOL6CcR91YHbqibEVF2aSk=";
+
+  meta = with lib; {
+    description = "M3U Proxy for Plex DVR and Emby Live TV";
+    homepage = "https://github.com/xteve-project/xTeVe";
+    license = licenses.mit;
+    maintainers = with maintainers; [ nrhelmi AndersonTorres ];
```
Comment (AndersonTorres) [inline pkgs/servers/xteve/default.nix:20]:
```suggestion
    maintainers = with maintainers; [ nrhelmi ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Again:
Your pull request has eight commits. Reduce it to only two:

1. The first commit adds yourself to maintainers list
2. The second effectively adds the package

You can do this by using `git rebase` facility.

**Comment by AndersonTorres** [comment]
> BTW is there any documentation on steps (commit policies) to add a new package to nixpkgs ?

What exactly you want to know? The barriers to add packages to Nixpkgs are very low when compared to other package databases. It just needs someone with free time to revise and test.

**Comment by AndersonTorres** [comment]
Third time:

Your pull request has nine commits. Reduce it to only two:

1. The first commit adds yourself to maintainers list
2. The second effectively adds the package

You can do this by using `git rebase` facility.

**Comment by AndersonTorres** [comment]
Ping.

**Comment by NRHelmi** [comment]
> Ping.

Hi @AndersonTorres sorry didn't get a chance to look into will do during this weekend

**Comment by NRHelmi** [comment]
@AndersonTorres  closing this PR and moving the changes to https://github.com/NixOS/nixpkgs/pull/213411

**Pull request closed without merging**